squad/api/ci.py

```diff
@@ -51,3 +51,42 @@ def submit_job(request, group_slug, project_slug, version, environment_slug):
 
     # return ID of test job
     return HttpResponse(test_job.id, status=201)
+
+
+@require_http_methods(["POST"])
+@csrf_exempt
+@auth
+def watch_job(request, group_slug, project_slug, version, environment_slug):
+    backend_name = request.POST.get('backend')
+    if backend_name is None:
+        return HttpResponseBadRequest("backend field is required")
+    backend = None
+    try:
+        backend = Backend.objects.get(name=request.POST.get('backend'))
+    except Backend.DoesNotExist:
+        return HttpResponseBadRequest("requested backend does not exist")
+
+    # project has to exist or request will result with 404
+    project = Project.objects.get(slug=project_slug, group__slug=group_slug)
+    if backend is None or project is None:
+        return HttpResponseBadRequest("malformed request")
+
+    # testjob_id points to the backend's test job
+    testjob_id = request.POST.get('testjob_id', None)
+
+    if testjob_id is None:
+        return HttpResponseBadRequest("testjob_id is required")
+
+    # create TestJob object
+    test_job = TestJob.objects.create(
+        backend=backend,
+        target=project,
+        build=version,
+        environment=environment_slug,
+        job_id=testjob_id
+    )
+    # schedule test job check
+    fetch.delay(test_job.id)
```

Comment:
I guess you're right. If the listener daemon only cares about job_id, it will still work without fetching details.